### PR TITLE
Convert to dapper

### PR DESCRIPTION
**Pull request by cloudnautique**
This converts the build/test/etc.. to Dapper and the Rancher `./scripts` directory format.

We will not build on tags with stable, but all other situations should build.
promote will only run on stable, but if we add more channels it is easy to adapt.

Should only upload promoted

### CODE REVIEW

File: scripts/test [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 set -e
 
+cd $(dirname $0)/..
+
+helm init -c
+helm plugin install https://github.com/lrills/helm-unittest
```
Comment (jgreat) [inline scripts/test]: I think Line 7 and 8 are really install process and should moved into the Dockerfile.  

Just to reduce on my silly errors I'm linking the test script as a pre-commit hook in git. How important is it to split out the `helm lint` into its own step?

File: Dockerfile.dapper [hunk 2]
```diff
@@ -0,0 +1,22 @@
+FROM debian
+
+ARG DAPPER_HOST_ARCH
+ENV HOST_ARCH=${DAPPER_HOST_ARCH} ARCH=${DAPPER_HOST_ARCH}
+ENV CATTLE_HELM_VERSION v2.10.0-rancher4
```
Comment (jgreat) [inline Dockerfile.dapper]: Not sure why this is here.  Looks like we're pulling latest from kubernetes/helm/master

**Comment by cloudnautique** [comment]
As far as lint in it's own step, we have convention in our other projects to have a validate script. 
so you can do `make validate` and `make test` 
and `make ci` executes all of the steps. 

It seems like the lint and unittests are fast enough together that we could merge them into `test`. If that's what you'd like, let me know and I'll adjust. It's not that big of a deal.

**Review by jgreat — COMMENTED** [review]
LGTM

**Pull request merged**